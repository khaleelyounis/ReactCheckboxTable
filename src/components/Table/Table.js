import React from 'react';
import PropTypes from 'prop-types';

const Table = () => {
  return (
    <table>
      <thead></thead>
      <tbody></tbody>
    </table>
  );
};

Table.defaultProps = {};

Table.propTypes = {};

export default Table;
