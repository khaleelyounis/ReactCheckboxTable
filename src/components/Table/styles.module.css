.tableHeaderRow {
  text-align: left;
  padding: 15px;
}

.tableHeaderData {
  padding: 1rem 0;
  font-weight: 500;
}

.tableBodyRow:hover {
  background-color: rgba(211, 211, 211, 0.2);
  cursor: pointer;
}

.tableBodyData {
  padding: 1rem 0;
  border-bottom: 0.5px solid rgb(169, 169, 169);
}

.tableHeaderCheckbox,
.tableBodyCheckbox {
  padding: 1rem 1rem 1rem 0.5rem;
  border-bottom: 0.5px solid rgb(169, 169, 169);
}

.tableBodyCheckbox:hover {
  cursor: pointer;
}

.greenIcon {
  height: 20px;
  width: 20px;
  float: right;
  padding-right: 0.5rem;
}
