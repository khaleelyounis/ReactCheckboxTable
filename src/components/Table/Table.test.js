import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Table from './Table';

Table.defaultProps = {
  tableHeader: ['header1', 'header2', 'header3'],
  tableData: [
    { header1: '1', header2: '2', header3: '3' },
    { header1: '4', header2: '5', header3: '6' },
  ],
  onClick: jest.fn(),
  isItemSelected: jest.fn(),
  selectedCount: 0,
  checked: false,
  downloadCallback: jest.fn(),
  toggleAll: jest.fn(),
  tableActions: true,
  selectable: true,
};

describe('Table Snapshots', () => {
  it('Should render with checkboxes and tableActions', () => {
    const component = render(<Table />);
    const checkboxes = component.getAllByTestId('checkbox');
    const tableActions = component.getByTestId('tableActions');

    expect(tableActions).toBeTruthy();
    expect(checkboxes).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
  it('Should render without checkboxes and tableActions', () => {
    const component = render(<Table tableActions={false} selectable={false} />);
    const checkboxes = component.queryByTestId('checkbox');
    const tableActions = component.queryByTestId('tableActions');

    expect(tableActions).toBeNull();
    expect(checkboxes).toBeNull();
    expect(component).toMatchSnapshot();
  });
});

describe('Table Actions', () => {
  it('Should fire onClick function when row is clicked', () => {
    const component = render(<Table />);
    const tableRow = component.getAllByTestId('tableBodyRow')[0];

    fireEvent.click(tableRow);
    expect(Table.defaultProps.onClick).toHaveBeenCalled();
  });

  it('Should fire onClick function when checkbox is clicked', () => {
    const component = render(<Table />);
    const checkbox = component.getAllByTestId('checkbox')[0];

    fireEvent.click(checkbox);
    expect(Table.defaultProps.onClick).toHaveBeenCalled();
  });
});
