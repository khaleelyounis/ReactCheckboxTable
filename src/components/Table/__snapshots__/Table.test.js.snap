// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Snapshots Should render with checkboxes and tableActions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tableActions"
        data-testid="tableActions"
      >
        <div
          class="tableActionsCheckbox"
          data-testid="tableActionsCheckbox"
        >
          <input
            data-testid="checkbox"
            name="checkbox"
            type="checkbox"
          />
           
        </div>
        <div
          class="tableActionsSelected"
          data-testid="tableActionsSelected"
        >
          None Selected
        </div>
        <div
          class="tableActionsDownload"
          data-testid="tableActionsDownload"
        >
          <img
            alt="download icon"
            class="downloadIcon"
            data-testid="downloadIcon"
            src="download.png"
          />
          Download Selected
        </div>
      </div>
      <table
        data-testid="table"
      >
        <thead
          data-testid="thead"
        >
          <tr
            class="tableHeaderRow"
            data-testid="tableHeaderRow"
          >
            <th
              class="tableHeaderCheckbox"
              data-testid="tableHeaderCheckbox"
            />
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header1
            </th>
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header2
            </th>
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header3
            </th>
          </tr>
        </thead>
        <tbody
          data-testid="tbody"
        >
          <tr
            class="tableBodyRow"
            data-testid="tableBodyRow"
          >
            <td
              class="tableBodyCheckbox"
              data-testid="tableBodyCheckbox"
            >
              <input
                data-testid="checkbox"
                name="checkbox"
                type="checkbox"
              />
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              1
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              2
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              3
            </td>
          </tr>
          <tr
            class="tableBodyRow"
            data-testid="tableBodyRow"
          >
            <td
              class="tableBodyCheckbox"
              data-testid="tableBodyCheckbox"
            >
              <input
                data-testid="checkbox"
                name="checkbox"
                type="checkbox"
              />
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              4
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              5
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              6
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <div
      class="tableActions"
      data-testid="tableActions"
    >
      <div
        class="tableActionsCheckbox"
        data-testid="tableActionsCheckbox"
      >
        <input
          data-testid="checkbox"
          name="checkbox"
          type="checkbox"
        />
         
      </div>
      <div
        class="tableActionsSelected"
        data-testid="tableActionsSelected"
      >
        None Selected
      </div>
      <div
        class="tableActionsDownload"
        data-testid="tableActionsDownload"
      >
        <img
          alt="download icon"
          class="downloadIcon"
          data-testid="downloadIcon"
          src="download.png"
        />
        Download Selected
      </div>
    </div>
    <table
      data-testid="table"
    >
      <thead
        data-testid="thead"
      >
        <tr
          class="tableHeaderRow"
          data-testid="tableHeaderRow"
        >
          <th
            class="tableHeaderCheckbox"
            data-testid="tableHeaderCheckbox"
          />
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header1
          </th>
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header2
          </th>
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header3
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="tbody"
      >
        <tr
          class="tableBodyRow"
          data-testid="tableBodyRow"
        >
          <td
            class="tableBodyCheckbox"
            data-testid="tableBodyCheckbox"
          >
            <input
              data-testid="checkbox"
              name="checkbox"
              type="checkbox"
            />
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            1
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            2
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            3
          </td>
        </tr>
        <tr
          class="tableBodyRow"
          data-testid="tableBodyRow"
        >
          <td
            class="tableBodyCheckbox"
            data-testid="tableBodyCheckbox"
          >
            <input
              data-testid="checkbox"
              name="checkbox"
              type="checkbox"
            />
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            4
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            5
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            6
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Table Snapshots Should render without checkboxes and tableActions 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        data-testid="table"
      >
        <thead
          data-testid="thead"
        >
          <tr
            class="tableHeaderRow"
            data-testid="tableHeaderRow"
          >
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header1
            </th>
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header2
            </th>
            <th
              class="tableHeaderData"
              data-testid="tableHeaderData"
            >
              header3
            </th>
          </tr>
        </thead>
        <tbody
          data-testid="tbody"
        >
          <tr
            class="tableBodyRow"
            data-testid="tableBodyRow"
          >
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              1
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              2
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              3
            </td>
          </tr>
          <tr
            class="tableBodyRow"
            data-testid="tableBodyRow"
          >
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              4
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              5
            </td>
            <td
              class="tableBodyData"
              data-testid="tableBodyData"
            >
              6
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      data-testid="table"
    >
      <thead
        data-testid="thead"
      >
        <tr
          class="tableHeaderRow"
          data-testid="tableHeaderRow"
        >
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header1
          </th>
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header2
          </th>
          <th
            class="tableHeaderData"
            data-testid="tableHeaderData"
          >
            header3
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="tbody"
      >
        <tr
          class="tableBodyRow"
          data-testid="tableBodyRow"
        >
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            1
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            2
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            3
          </td>
        </tr>
        <tr
          class="tableBodyRow"
          data-testid="tableBodyRow"
        >
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            4
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            5
          </td>
          <td
            class="tableBodyData"
            data-testid="tableBodyData"
          >
            6
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
